relay-server/src/endpoints/project_configs.rs

```diff
@@ -122,6 +128,15 @@ async fn inner(
 
     let mut configs = HashMap::with_capacity(keys_len);
     let mut pending = Vec::with_capacity(keys_len);
+    let global = if inner.global_config {
+        global_config_service
+            .send(GetGlobalConfig)
+            .await
+            .ok()
+            .map(|gc| (*gc).clone())
+    } else {
+        None
+    };
```

Comment:
Does `then` make this cleaner? Something like this (haven't tried it):
```suggestion
    let global = inner.global_config.then(|| {
        global_config_service
            .send(GetGlobalConfig)
            .await
            .ok()
            .map(|gc| (*gc).clone())
    });
```

Reply:
i tried a while before i gave up on that approach haha. the main issue is the async within the closure, maybe it's possible but I didn't succeed